src/uucore/src/lib/features/checksum.rs

```diff
@@ -68,82 +68,68 @@ pub struct HashAlgorithm {
     pub bits: usize,
 }
 
-#[derive(Debug)]
-pub enum ChecksumError {
-    RawMultipleFiles,
-    IgnoreNotCheck,
-    InvalidOutputSizeForSha3,
-    BitsRequiredForSha3,
-    BitsRequiredForShake128,
-    BitsRequiredForShake256,
-    UnknownAlgorithm,
-    InvalidLength,
-    LengthOnlyForBlake2b,
-    BinaryTextConflict,
-    AlgorithmNotSupportedWithCheck,
-    CombineMultipleAlgorithms,
-    NeedAlgorithmToHash,
-    NoProperlyFormattedChecksumLinesFound(String),
+struct ChecksumResult {
+    pub bad_format: i32,
+    pub failed_cksum: i32,
+    pub failed_open_file: i32,
+    pub correct_format: i32,
+    pub properly_formatted: bool,
 }
```

Comment:
I'm struggling with this struct and I think `correct_format` and `properly_formatted` don't fit in. The first three properties are failure counters shown to the user and the other two are something else. Plus name and type of `correct_format` are a bit misleading because the property is used as a bool.